### ISSUE TITLE
feature request: CNI file permissions more restrictive (600)

### ISSUE THREAD

**Issue by moonek** [projectcalico/calico#7461]
## Expected Behavior

CNI-related CIS Benchmarks include:
```
1.1.9	Ensure that the Container Network Interface file permissions are set to 600 or more restrictive
1.1.10	Ensure that the Container Network Interface file ownership is set to root:root
```

## Current Behavior

`/etc/cni/net.d/10-calico.conflist` file permission is 644.
```
root@master:/root$ ll /etc/cni/net.d/
total 8
drwxr-xr-x. 2 root root   57 Nov 16 05:33 ./
drwxr-xr-x. 3 root root   19 Apr 18  2022 ../
-rw-r--r--  1 root root  657 Mar 13 03:46 10-calico.conflist
-rw-------  1 root root 2712 Mar 13 03:46 calico-kubeconfig
```

## Steps to Reproduce (for bugs)

1. curl https://raw.githubusercontent.com/projectcalico/calico/v3.24.5/manifests/calico.yaml -O
2. kubectl apply -f calico.yaml
3. ll /etc/cni/net.d/
4.

## Your Environment

* Calico version: 3.24.5
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes 1.24.8
* Operating System and version: Centos 7.8
* Link to your project (optional):

**Comment by moonek** [comment]
After installation, I wonder if it is okay to manually change the above file(`/etc/cni/net.d/10-calico.conflist`) permissions to 600.

**Comment by mgleung** [comment]
@moonek I don't particularly remember if we had a reason to have the current 644 permissions. It sounds like this is mostly a matter of testing so if you or any other community members would like to help us test out those permissions, let us know!

**Comment by caseydavenport** [comment]
Looks like our install script maintains the permissions from the binaries packaged in the docker image: https://github.com/projectcalico/calico/blob/master/cni-plugin/pkg/install/install.go#L215

Should just be a matter of changing the permissions as part of the cni-plugin image build here: https://github.com/projectcalico/calico/blob/master/cni-plugin/Makefile#L132-L141

**Comment by missa-wndrvr** [comment]
Hello, would this feature be considered for the next release?

**Comment by kron4eg** [comment]
@caseydavenport has this been fixed?

**Comment by caseydavenport** [comment]
Yep, this PR went in to v3.30: https://github.com/projectcalico/calico/pull/8991
